src/sql/workbench/parts/query/common/queryInput.ts

```diff
@@ -130,43 +141,28 @@ export class QueryInput extends EditorInput implements IEncodingSupport, IConnec
 			this._register(this._sql.onDidChangeDirty(() => this._onDidChangeDirty.fire()));
 		}
 
-		// Attach to event callbacks
-		if (this._queryModelService) {
-			// Register callbacks for the Actions
-			this._register(
-				this._queryModelService.onRunQueryStart(uri => {
-					if (this.uri === uri) {
-						this.onRunQuery();
-					}
-				})
-			);
-
-			this._register(
-				this._queryModelService.onRunQueryComplete(uri => {
-					if (this.uri === uri) {
-						this.onQueryComplete();
-					}
-				})
-			);
-		}
+		// Register callbacks for the Actions
+		this.runner.onQueryStart(() => this.onRunQuery());
+
```

Comment:
Not registering this for disposal?

Reply:
the runner is registered, so all its emitters are registered.